### DISCUSSION TITLE
Add badges to resources in the grid view

### DISCUSSION THREAD

**Discussion by iainbeeston** [avo-hq/avo#2736] [Feature requests]
It would be really useful if I could add an icon that appears in the corner of the images on the grid view, to show the status of that item. That would make it much easier to visually see the status of items in a grid without having to go into the show or edit views (this isn't as much of an issue with the table view because you can add as many columns as you need). Something like this might work?

```ruby
self.grid_view = {
    card: -> do
      {
        badge: {
          position: "top-right",
          icon: "heroicons/outline/academic-cap",
          color: "#00ff00",
      }
    end
  }
  ```

Which might look like this:

![Screenshot 2024-05-02 at 16 24 25](https://github.com/avo-hq/avo/assets/111963/e8c0e954-c5a1-4366-8236-f96e19f36bd5)

**Comment by adrianthedev** [comment]
This is a great idea!
We'd merge something like this if we had a PR.

BTW, there's this PR (https://github.com/avo-hq/avo/pull/2650) that I need to finish.
Would this make the badges (this request) obsolete?

**Comment by iainbeeston** [comment]
I think #2650 would be very helpful and I’d use that if it was merged, but I think if I could choose between that and a badge then I would prefer a badge. I don’t have the time to work on the badges, but if you finish #2650 I would use that on my project.

**Comment by Paul-Bob** [comment]
https://docs.avohq.io/3.0/grid-view.html#grid-item-badge
